posthog/heatmaps/sql.py

```diff
@@ -0,0 +1,192 @@
+from django.conf import settings
+
+from posthog.clickhouse.kafka_engine import kafka_engine
+from posthog.clickhouse.table_engines import (
+    Distributed,
+    ReplicationScheme,
+    MergeTreeEngine,
+)
+from posthog.kafka_client.topics import KAFKA_CLICKHOUSE_HEATMAP_EVENTS
+
+HEATMAPS_DATA_TABLE = lambda: "sharded_heatmaps"
+
+
+"""
+We intend to send specific $heatmap events to build a heatmap instead of building from autocapture like the click map
+We'll be storing individual clicks per url/team/session
+And we'll be querying for those clicks at day level of granularity
+And we'll be querying by URL exact or wildcard match
+And we'll _sometimes_ be querying by width
+
+We _could_ aggregate this data by day, but we're hoping this will be small/fast enough not to bother
+And can always add a materialized view for day (and week?) granularity driven by this data if needed
+
+We only add session_id so that we could offer example sessions for particular clicked areas in the toolbar
+"""
+
+KAFKA_HEATMAPS_TABLE_BASE_SQL = """
+CREATE TABLE IF NOT EXISTS {table_name} ON CLUSTER '{cluster}'
+(
+    session_id VARCHAR,
+    team_id Int64,
+    distinct_id VARCHAR,
+    timestamp DateTime64(6, 'UTC'),
+    -- x is the x with resolution applied, the resolution converts high fidelity mouse positions into an NxN grid
+    x Int16,
+    -- y is the y with resolution applied, the resolution converts high fidelity mouse positions into an NxN grid
+    y Int16,
+    -- stored so that in future we can support other resolutions
+    scale_factor Int16,
+    viewport_width Int16,
+    viewport_height Int16,
+    -- some elements move when the page scrolls, others do not
+    pointer_target_fixed Bool,
+    current_url VARCHAR,
+    type LowCardinality(String)
+) ENGINE = {engine}
+"""
+
+HEATMAPS_TABLE_BASE_SQL = """
+CREATE TABLE IF NOT EXISTS {table_name} ON CLUSTER '{cluster}'
+(
+    session_id VARCHAR,
+    team_id Int64,
+    distinct_id VARCHAR,
+    timestamp DateTime64(6, 'UTC'),
+    -- x is the x with resolution applied, the resolution converts high fidelity mouse positions into an NxN grid
+    x Int16,
+    -- y is the y with resolution applied, the resolution converts high fidelity mouse positions into an NxN grid
+    y Int16,
+    -- stored so that in future we can support other resolutions
+    scale_factor Int16,
+    viewport_width Int16,
+    viewport_height Int16,
+    -- some elements move when the page scrolls, others do not
+    pointer_target_fixed Bool,
+    current_url VARCHAR,
+    type LowCardinality(String),
+    _timestamp DateTime
```

Comment:
something else that would be _lovely_ to add here are the following [virtual columns](https://clickhouse.com/docs/en/engines/table-engines/integrations/kafka#virtual-columns):
`_offset`
`_partition`

We've really wished to have these on tables in the past - would be super cool to start adding them here.